infrastructure-playbooks/switch-from-non-containerized-to-containerized-ceph-daemons.yml

```diff
@@ -281,24 +274,22 @@
 
     - name: check if containerized osds are already running
       shell: |
-        docker ps | grep -sq {{ item | regex_replace('/', '') }}
+        docker ps | grep -sqE ceph-osd-[0-9]
       changed_when: false
       failed_when: false
-      with_items: "{{ devices }}"
       register: osd_running
 
-    - name: resolve device(s) path(s)
-      command: readlink -f {{ item }}
+    - name: get osd directories
+      command: ls -1 /var/lib/ceph/osd
+      register: osd_dirs
       changed_when: false
-      with_items: "{{ devices }}"
-      register: devices_canonicalize
 
     - name: unmount all the osd directories
-      command: umount "{{ item.0.stdout }}"1
+      command: umount "{{ item.0.stdout }}"
       changed_when: false
       failed_when: false
       with_together:
-        - "{{ devices_canonicalize.results }}"
+        - "{{ osd_dirs.stdout }}"
```

Comment:
`{{ osd_dirs.stdout }}` is a string like :

`ceph-osd1\nceph-osd2\nceph-osd3` right?

don't you want `osd_dirs.stdout_lines` which is an array like `['ceph-osd1', 'ceph-osd2', 'ceph-osd3']` instead ?